### ISSUE TITLE
Snippets, Gists, Etc. feature

### ISSUE THREAD

**Issue by robmoggach** [gogits/gogs#936]
Would be great to see a snippets feature added like github gists or gitlab snippets... It's an invaluable tool when you don't want a full blown repo.

**Comment by Unknwon** [comment]
Thanks your feature request!

We want users to be able edit in the web UI, so this feature may come together/after https://github.com/gogits/gogs/issues/270

**Comment by MichielDeMey** [comment]
You could potentially integrate with GitHub's gists for this.

They don't charge you for them and allow you to create an unlimited amount.

However I can understand that not everyone wants to give their data to GitHub.

**Comment by robmoggach** [comment]
the feature request was to add a feature like github gists or gitlab snippets, not supplant it using one of those existing solutions by other providers - ie. add it as a feature to gogs - I already use gists regularly and see it as a great feature for gogs

**Comment by MichielDeMey** [comment]
I am aware of what the request was. I'm just not clear on the added benefit of adding this to Gogs versus using an existing (great) solution.

The only thing I can think of is owning your own data, which I agree can be useful in some instances.

**Comment by robmoggach** [comment]
Not sure why this isn't obvious...

-private intranet
-data ownership
-self imposed security policies
-whole stack management of IT

**Comment by StrangeWill** [comment]
Pretty much agree with @robmoggach, if we wanted to use GitHub's Gists, we'd just use them. Gogs can end up on a system behind a firewall that can't push things to GitHub (one of the benefits of a self-hosted solution), so of course this needs to be standalone.

**Comment by robmoggach** [comment]
thanks... shouldn't really be a discussion but good to have some backup.
GitLab does it... gogs should too.

--
Robert Moggach
Sr. VFX Supervisor / Flame Artist
Los Angeles / New York / London / Toronto
http://moggach.com
https://www.linkedin.com/profile/view?id=1664843
Mobile (310) 773-1793

**Comment by kolis** [comment]
This would be a really awesome feature, nice to see it in a milestone :D

**Comment by Unknwon** [comment]
@kolis :smile:

**Comment by mkoitka** [comment]
We would also use this feature internally. It'd be nice if multiple "files" could be added to a snippet like Github does with its Gists. That would also be a real advantage over GitLab. IMHO they only allow adding of one file per snippet.

**Comment by joubertredrat** [comment]
:+1:  +1 with love

**Comment by NiklasRosenstein** [comment]
+1

**Comment by nmaas87** [comment]
+1 :)

**Comment by xxxtonixxx** [comment]
+1 :+1:

**Comment by asmartin** [comment]
+1

**Comment by mehrad-rafigh** [comment]
+1. I need to learn Go man :)

**Comment by SirCmpwn** [comment]
+1

**Comment by j4** [comment]
+1

**Comment by peponi** [comment]
+1

**Comment by stephanfriedrich** [comment]
+1

**Comment by jgreth** [comment]
+1

**Comment by Jitsusama** [comment]
+1

**Comment by Einfach-Ich** [comment]
+1

**Comment by lavvy** [comment]
Any news about this topic

**Comment by PascalBru** [comment]
+1

**Comment by hhenkel** [comment]
As #270 has been implemented in december 2015 and  the web editor is available since a few days, it might be good to look at this feature again?

**Comment by BryanDollery** [comment]
Any news on this? It would be very useful.

**Comment by andyph666** [comment]
+1

**Comment by FoRTu** [comment]
+1

Any info about it?

Thank you!

**Comment by yanlinaung30** [comment]
Any update for gist?

**Comment by newey01c** [comment]
:+1: would be really great to see this feature

**Comment by Fastidious** [comment]
Everyone seems to be expecting this (it would be a very nice addition!), but no one is sending in any PR to contribute. I do not have knowledge of `go`, but if anyone does, let's stop with the `+1`, and short inquires, and just sent in your PRs.

Please, learn how to use issues on GitHub. This isn't Facebook, nor Twitter. Your `+1` only comments do nothing else but to pollute the original request.

**Comment by The5heepDev** [comment]
@Fastidious agreed. I think they are trying to bump the issue so that devs would see it, but it is very annoying. 

I am the owner/creator of https://comp.st (which itself is written in go)

**Comment by dbluhm** [comment]
I can't say I know how github stores gists but I personally think storing them in a single git repo would be good for exactly the reason you mentioned. Batch download as well as the potential to just clone your gists locally would be compelling features -- for me, at least.

In the end, the only difference between the gists repo and your other repos would be how it's presented.

**Comment by Fastidious** [comment]
> I could either store it in a fs based db lke comp.st, or I could store gists in a single git repo (which could be useful for batch downloading a user's gist

Storing gists in a single repository would be the way to go.  How goes progress, @The5heepDev?

**Comment by The5heepDev** [comment]
@Fastidious hey sorry I've been swamped lately. I'm setting up a testing environment right now, though!

**Comment by The5heepDev** [comment]
@Fastidious I have a (somewhat hacky) setup for pastes on my fork. 
for the pastes to work you need to create a repo named 'paste', with some file inside it. It cannot be a blank repo.

https://github.com/The5heepDev/gogs/tree/develop

**Comment by Fastidious** [comment]
@The5heepDev it is a start!

**Comment by The5heepDev** [comment]
I currently am just using the repo templates with `{{if not .IsPasteRepo}}{{end}}` to remove some elements that don't pertain to the pastes. The next step is to move the testing scripts and stuff into it's own route pkg and dedicated templates. For now, much of it will be recycled from the existing repo template/pkg

**Comment by PascalBru** [comment]
any progress

**Comment by raniellyferreira** [comment]
+1

**Comment by CWempe** [comment]
Any news on this one?

**Comment by ttimasdf** [comment]
+1, another ping in case it get lost in 2021 😃

**Comment by apeyroux** [comment]
👍

**Comment by niemes** [comment]
+1

**Comment by Michal-Szczepaniak** [comment]
+1

**Comment by r1der23** [comment]
+1

**Comment by h0m3** [comment]
+1

**Comment by gdog2u** [comment]
+1 As github gets worse and worse, I hope this can be added soon

**Comment by ulissesjdeo** [comment]
+1 It would be awesome!

**Comment by Michal-Szczepaniak** [comment]
@ulissesjdeo first +1 in 2024 since 2022 so i'll repeat my +1 just to reassure the devs
